File: CSharp/Enterprise/ObjectRelational/Repository/Product.cs
using System;

namespace Enterprise.ObjectRelational.Repository;

/// <summary>
/// Another example domain entity.
/// </summary>
public class Product : IEntity
{
    public int Id { get; set; }
    public string Name { get; set; }
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public string Category { get; set; }

    public Product(string name, decimal price, int stockQuantity, string category)
    {
        Name = name ?? throw new ArgumentNullException(nameof(name));

        if (price < 0)
            throw new ArgumentException("Price cannot be negative", nameof(price));

        if (stockQuantity < 0)
            throw new ArgumentException("Stock quantity cannot be negative", nameof(stockQuantity));

        Price = price;
        StockQuantity = stockQuantity;
        Category = category ?? throw new ArgumentNullException(nameof(category));
    }

    public override string ToString()
    {
        return $"Product[{Id}]: {Name} - ${Price} (Stock: {StockQuantity}, Category: {Category})";
    }
}

File: Java/Enterprise/Base/Product.java
package Enterprise.Base;

/**
 * Example domain entity extending Layer Supertype
 */
public class Product extends DomainObject {
    private String name;
    private double price;

    public Product(String name, double price) {
        super();
        this.name = name;
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
        markModified();
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
        markModified();
    }

    @Override
    public String toString() {
        return "Product{id=" + getId() + ", name='" + name + "', price=" + price + "}";
    }
}
